var/spack/repos/builtin/packages/gasnet/package.py

```diff
@@ -6,98 +6,133 @@
+
+    def conduits(self):
+        return [c for c in ['smp', 'mpi', 'ibv', 'udp'] if "+" + c in self.spec]
+
+    def install(self, spec, prefix):
+        if spec.satisfies('@master:'):
+            bootstrapsh = Executable("./Bootstrap")
+            bootstrapsh()
+
+        # The GASNet-EX library has a highly multi-dimensional configure space,
+        # to accomodate the varying behavioral requirements of each client runtime.
+        # The library's ABI/link compatibility is strongly dependent on these
+        # client-specific build-time settings, and that variability is deliberately NOT
+        # encoded in the variants of this package. The recommended way to build/deploy
+        # GASNet is as an EMBEDDED library within the build of the client package
+        # (eg. Berkeley UPC, UPC++, Legion, etc), some of which provide build-time
+        # selection of the GASNet library sources. This spack package provides
+        # the GASNet-EX sources, for use by appropriate client packages.
+        install_tree('.', prefix + "/src")
+
+        # Library build is provided for unit-testing purposes only (see notice above)
+        if len(self.conduits()) > 0:
+            options = ["--prefix=%s" % prefix]
+
+            if '+debug' in spec:
+                options.append("--enable-debug")
+
+            if '+mpi' in spec:
+                options.append("--enable-mpi-compat")
+            else:
+                options.append("--disable-mpi-compat")
+
+            options.append("--disable-auto-conduit-detect")
+            for c in self.conduits():
+                options.append("--enable-" + c)
+
+            configure(*options)
+            make()
+            make('install')
+
+            for c in self.conduits():
+                testdir = join_path(prefix, 'tests', c)
+                mkdirp(testdir)
+                make('-C', c + '-conduit', 'testgasnet-par')
+                install(c + "-conduit/testgasnet", testdir)
+            make('-C', c + '-conduit', 'testtools-par')
+            install(c + "-conduit/testtools", join_path(prefix, 'tests'))
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def test_install(self):
+        if '+smp' in self.spec:
+            make('-C', 'smp-conduit', 'run-tests')
+        if len(self.conduits()) > 0:
+            self.run_test(join_path(prefix, 'tests', 'testtools'),
+                          expected=['Done.'], status=0,
+                          installed=True, purpose="Running testtools")
+
+    def test(self):
+        ranks = '4'
+        spawner = {
+            'smp': ['env', 'GASNET_PSHM_NODES=' + ranks],
+            'mpi': [join_path(self.prefix.bin, 'gasnetrun_mpi'), '-n', ranks],
+            'ibv': [join_path(self.prefix.bin, 'gasnetrun_ibv'), '-n', ranks],
+            'udp': [join_path(self.prefix.bin, 'amudprun'), '-spawn', 'L', '-np', ranks]
+        }
+
+        env['GASNET_VERBOSEENV'] = '1'  # include diagnostic info
```

Comment:
FWIW. I tend to prefer being explicit wrt the environment (`os.environ`) as well but am probably pushing it in this case. :grin:

Reply:
Fixed as suggested in 83f9bf4